src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -8500,6 +8500,84 @@ double SketchObject::calculateAngleViaPoint(int GeoId1, int GeoId2, double px, d
         throw Base::ValueError("Null geometry in calculateAngleViaPoint");
 }
 
+bool SketchObject::calculateAngle(int& GeoId1, int& GeoId2, Sketcher::PointPos& PosId1, Sketcher::PointPos& PosId2, double& ActAngle)
+{
+    const Part::Geometry* geom1 = getGeometry(GeoId1);
+    const Part::Geometry* geom2 = getGeometry(GeoId2);
+
+    if (geom1->getTypeId() == Part::GeomLineSegment::getClassTypeId() &&
+        geom2->getTypeId() == Part::GeomLineSegment::getClassTypeId()) {

```

Comment:
You changed the behaviour when trying to make the guard clause. You didn't invert the condition

Reply:
Ah this is unfortunate. Fixed